lib/Doctrine/ORM/Query/SqlWalker.php

```diff
@@ -1962,13 +1737,10 @@ public function walkWhereClause($whereClause)
 
     /**
      * Walk down a ConditionalExpression AST node, thereby generating the appropriate SQL.
-     *
-     * @param AST\ConditionalExpression $condExpr
-     *
-     * @return string
      */
-    public function walkConditionalExpression($condExpr)
-    {
+    public function walkConditionalExpression(
+        AST\ConditionalExpression|AST\ConditionalPrimary|AST\ConditionalTerm|AST\ConditionalFactor $condExpr
```

Comment:
I'm not really happy with those unions here. We should try to get rid of them in a follow-up.